src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,64 @@
+package duke.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * The Deadline class represents a task associated with a deadline.
+ * The instance of the Deadline class has the description and
+ * the time of the deadline.
+ */
+public class Deadline extends Task {
+    protected LocalDate by;
+
+    /**
+     * Takes in the description and time of a deadline and returns an instance of
+     * the corresponding type of task.
+     *
+     * @param description The description of the deadline.
+     * @param by The time of the deadline.
+     */
+    public Deadline(String description, LocalDate by) {
+        super(description);
+        this.by = by;
+    }
+
+    /**
+     * Returns a representation of the task in terms of the description and state.
+     *
+     * @return The representation of the task.
+     */
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + by.format(DateTimeFormatter.ofPattern("MMM d yyyy")) + ")";
+    }
+
+    /**
+     * Returns a representation of the task in terms of the description and
+     * state for the task to be stored into the record.
+     *
+     * @return The representation of the task in terms of
+     *         the description, the state of completion, and time.
+     */
+    @Override
+    public String record() {
+        if (this.isCompleted) {
+            return "D | 1 | " + description + " | " + by;
```

Comment:
(just a suggestion) perhaps you could store the tasks in the same format as when the user inputs them/when you print the message on the console?